controllers/storageclassclaim_controller.go

```diff
@@ -572,14 +578,18 @@ func (r *StorageClassClaimReconciler) own(resource metav1.Object) error {
 	return controllerutil.SetOwnerReference(r.storageClassClaim, resource, r.Scheme)
 }
 
-// addAnnotation add a annotation to a resource metadata
-func addAnnotation(obj metav1.Object, key string, value string) {
+// addAnnotation add an annotation to a resource metadata, returns true if added
+func addAnnotation(obj metav1.Object, key string, value string) bool {
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
 		annotations = map[string]string{}
 		obj.SetAnnotations(annotations)
 	}
-	annotations[key] = value
+	if annotations[key] != value {
+		annotations[key] = value
+		return true
+	}
+	return false
```

Comment:
This will give the wrong answer if there was no annotation and you are trying to set an annotation with a value of an empty string

```suggestion
	oldValue, ok := annotations[key]
	if return !ok || oldValue != value {
		annotations[key] = value
		return true
	}
	return false
```

Reply:
done.